components/writers/src/main/java/org/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -129,40 +147,37 @@ public void configureForTransformedData(AnalysisJobBuilder ajb, TransformerDescr
     }
 
     @Override
-    public String getSuggestedLabel() {
-        if (file == null) {
-            return null;
-        }
-        return file.getName();
-    }
-
-    @Override
     public OutputWriter createOutputWriter() {
-        List<String> headers = new ArrayList<String>() ;
+        
+        if (file.exists() && overwriteFile==false){
```

Comment:
This condition could be moved up to a Validate method insted of the createOutputWriter() method (which is invoked at initiatlize time, whereas validation time is much more frequent/early).